### PR TITLE
populate enrollment test

### LINKED ISSUE #62: Create data migration with 2 English assessments

### PR DESCRIPTION
**Pull request by lemontree210**
closes #62

### CODE REVIEW

File: api/migrations/0002_data_migration.py [hunk 1]
```diff
@@ -97,6 +99,66 @@ def _write_languages_and_levels(self):
         )
         LanguageAndLevel.objects.bulk_create(language_and_level_objects)
 
+    def _write_enrollment_tests(self):
+        """Writes `EnrollmentTest', `Question`, and `EnrollmentTestQuestionOption` to database."""
```
Comment (Roxe322) [inline api/migrations/0002_data_migration.py:103]: Something wrong with quotes here :)

File: api/migrations/0002_data_migration.py [hunk 2]
```diff
@@ -97,6 +99,66 @@ def _write_languages_and_levels(self):
...
+                    enrollment_test=enrollment_test, text=item["text"]
+                )
+
+                # make sure exactly one option is marked as correct in each question
+                assert (
+                    len([o for o in item["options"] if o["is_correct"] is True]) == 1
+                ), f"Exactly 1 option has to be marked as correct for {item=}"
+
+                for option_data in item["options"]:
+                    EnrollmentTestQuestionOption.objects.create(
```
Comment (Roxe322) [inline api/migrations/0002_data_migration.py:149]: How about bulk_create?

File: api/migrations/0002_data_migration.py [hunk 3]
```diff
@@ -97,6 +99,66 @@ def _write_languages_and_levels(self):
...
+            / "migrations"
+            / "data_migration_sources"
+            / "enrollment_tests.yaml"
+        )
+        with path.open(encoding="utf8") as fh:
+            data = yaml.safe_load(fh)
+
+        for block in data:
+            enrollment_test = EnrollmentTest.objects.create(
+                language=Language.objects.get(id=block["language"]),
```
Comment (Roxe322) [inline api/migrations/0002_data_migration.py:125]: I think we can skip object getting by setting language_id directly

File: api/migrations/0002_data_migration.py (hunk 2 above)
Comment (lemontree210) [reply api/migrations/0002_data_migration.py]: Yes, fixed

File: api/migrations/0002_data_migration.py (hunk 1 above)
Comment (lemontree210) [reply api/migrations/0002_data_migration.py]: Yes :)

File: api/migrations/0002_data_migration.py (hunk 3 above)
Comment (lemontree210) [reply api/migrations/0002_data_migration.py]: Right, since id=locale for Language. Fixed

File: api/migrations/0002_data_migration.py [hunk 4]
```diff
@@ -97,6 +99,66 @@ def _write_languages_and_levels(self):
...
+                    )
+                    for option_data in item["options"]
+                ] + [  # add "don't know" to each question
+                    EnrollmentTestQuestionOption(
+                        question=question,
+                        text="(I don't know 😕)",
+                        is_correct=False,
+                    )
+                ]
+                EnrollmentTestQuestionOption.objects.bulk_create(options)
```
Comment (Roxe322) [inline api/migrations/0002_data_migration.py:160]: I think you can move this outside the for-loop, to be called only once. Also I'm not sure about question, but maybe it's also could be put into list for future bulk_create.

**Review by Roxe322 — APPROVED** [review]

File: api/migrations/0002_data_migration.py (hunk 4 above)
Comment (lemontree210) [reply api/migrations/0002_data_migration.py]: Done for options (don't think can be done for questions)

**Pull request merged**